### PR TITLE
[4.0] Missing icon on Crop button

### PR DESCRIPTION
**Pull request by N6REJ**
Pull Request for Issue #30024

### Summary of Changes
removed button- from button render as the class shouldn't be there

### Testing Instructions
Go to /administrator/index.php?option=com_templates&view=templates
Cassiopeia Details and Files
Click template_thumbnail.png

### Actual result BEFORE applying this Pull Request
Missing icon on Crop button
image

### Expected result AFTER applying this Pull Request
image
image

### Documentation Changes Required

none

### CODE REVIEW

File: libraries/src/Toolbar/ToolbarButton.php [hunk 1]
```diff
@@ -194,7 +194,7 @@ protected function renderButton(array &$options): string
 		);
 
 		$options['htmlAttributes'] = ArrayHelper::toString($options['attributes']);
-		$options['btnClass'] = 'button-' . $this->getName() . ' ' . ($options['btnClass'] ?? '');
+		$options['btnClass'] = ($options['btnClass'] ?? '');
```
Comment (Quy) [inline libraries/src/Toolbar/ToolbarButton.php]:
```suggestion
		$options['btnClass'] = $options['btnClass'] ?? '';
```

**Comment by N6REJ** [comment]
@Quy any idea whats creating the space before `btn btn-primary?`

**Comment by joeforjoomla** [comment]
Strange that there is a space before btn btn-primary, however removing this class could have side effects if the core or certain third-party extensions is using it to target a certain button... not sure if it's a good idea

**Comment by Quy** [comment]
See https://github.com/N6REJ/joomla-cms/blob/9256b24b9f702318831276790596eb6af96bfdbe/libraries/src/Toolbar/Button/StandardButton.php#L94

**Comment by N6REJ** [comment]
> Strange that there is a space before btn btn-primary, however removing this class could have side effects if the core or certain third-party extensions is using it to target a certain button... not sure if it's a good idea

I hear your concern @joeforjoomla .
do we want to encourage bad class usage?  The only other alternative is to rewrite the "standard" toolbar system completely.
I'm certainely open to suggestions.

**Comment by joeforjoomla** [comment]
@N6REJ having a  button- class in the button element doesn't seem a bad class usage to me. If you think so... i don't care

**Comment by N6REJ** [comment]
> @N6REJ having a button- class in the button element doesn't seem a bad class usage to me. If you think so... i don't care

I really don't think the attitude is necessary nor warranted.  The class can be added IF NEEDED/DESIRED by the application simply by supplying an "button-icon-class" afaik.
to have button- there with the current J! core will require a rewrite of the toolbar class(s).

**Comment by N6REJ** [comment]
we could add a variable $button_class and make it the class for the `<button>` itself.
if we stripped the icon- and fa- from $icon then we could use 
`$button_class . ' ' .  $icon` in the toolbar class and that would satisfy all uses I would think.  
Whats your thoughts?

**Comment by N6REJ** [comment]
@infograf768 This pr "fixes" the issue but in the process of fixing it I discovered another issue.
if we create the call w/o "fas fa-crop" and instead use NAME only.. such as
`ToolbarHelper::custom('template.cropImage', 'crop', 'arrows-alt', 'COM_TEMPLATES_BUTTON_CROP', false);`
then the problem goes away and all is well ( after this pr ) 
BUT if instead we use 
`ToolbarHelper::custom('template.cropImage', 'fas fa-crop', 'fas fa-arrows-alt', 'COM_TEMPLATES_BUTTON_CROP', false);`
then the toolbar id has a space in it.
image

**Comment by N6REJ** [comment]
also icon-over breaks the whole thing if added back into custom() like it should be.. as currently written its never used.

**Comment by N6REJ** [comment]
imo its ready for testing now.  There are design issues with icons but that has nothing to do with this issue/pr.

**Comment by N6REJ** [comment]
@joeforjoomla you'll be happy to know the button class is back

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 6f09d50571d05eb9982ac3c658bcdfa925b3f086<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30206">issues.joomla.org/tracker/joomla-cms/30206</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 6f09d50571d05eb9982ac3c658bcdfa925b3f086<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30206">issues.joomla.org/tracker/joomla-cms/30206</a>.</sub>

**Comment by jwaisner** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30206">issues.joomla.org/tracker/joomla-cms/30206</a>.</sub>

File: libraries/src/Toolbar/ToolbarButton.php [hunk 2]
```diff
@@ -211,7 +215,8 @@ protected function renderButton(array &$options): string
 	 */
 	protected function fetchId()
 	{
-		return $this->parent->getName() . '-' . $this->getName();
+
```
Comment (infograf768) [inline libraries/src/Toolbar/ToolbarButton.php]: No need for empty line here.

**Comment by infograf768** [comment]
Tks

**Pull request merged**